### ISSUE TITLE
Error message after installation on macOS.

### ISSUE THREAD

**Issue by M-Rick** [prrvchr/OAuth2OOo#15]
I cannot manage to install this extension.
I constantly get this error message:

`(com.sun.star.uno.RuntimeException) { { Message = "<class 'ImportError'>: cannot import name 'getAccessToken' from 'oauth2' (/Applications/LibreOffice.app/Contents/Frameworks/LibreOfficePython.framework/Versions/Current/lib/python3.8/site-packages/oauth2/__init__.py), traceback follows\X000a  File \"/Applications/LibreOffice.app/Contents/Resources/pythonloader.py\", line 147, in writeRegistryInfo\X000a    mod = self.getModuleFromUrl( locationUrl )\X000a  File \"/Applications/LibreOffice.app/Contents/Resources/pythonloader.py\", line 102, in getModuleFromUrl\X000a    exec(codeobject, mod.__dict__)\X000a  File \"/Users/aymeric/Library/Application Support/LibreOffice/4/user/uno_packages/cache/uno_packages/lu19028nx2d48.tmp_/OAuth2OOo.oxt/service/OAuth2Service.py\", line 57, in <module>\X000a    from oauth2 import getAccessToken\X000a\X000a at /Users/cloph/build/workdir/UnoApiHeadersTarget/udkapi/normal/com/sun/star/uno/RuntimeException.hdl:24", Context = (com.sun.star.uno.XInterface) @0 } }`

Version: 24.2.2.2 (AARCH64) / LibreOffice Community
Build ID: d56cc158d8a96260b836f100ef4b4ef25d6f1a01
CPU threads: 8; OS: macOS 14.4; UI render: Skia/Raster; VCL: osx
Locale: fr-FR (fr_FR.UTF-8); UI: fr-FR
Calc: threaded

**Comment by prrvchr** [comment]
Hi M-Rick,

I doubt that you have installed the latest version since the error message you give me does not point to the correct content.
- Error message: `OAuth2Service.py  line 57  from oauth2 import getAccessToken`
- Content: https://github.com/prrvchr/OAuth2OOo/blob/master/source/OAuth2OOo/service/OAuth2Service.py#L57

Could you confirm ?

**Comment by M-Rick** [comment]
Yes they are.

<img width="730" alt="Capture d’écran 2024-05-02 à 18 59 12" src="https://github.com/prrvchr/OAuth2OOo/assets/2520948/c182d1c4-9f8c-4dc0-bc85-b31284e7098a">

<img width="790" alt="Capture d’écran 2024-05-02 à 18 58 17" src="https://github.com/prrvchr/OAuth2OOo/assets/2520948/00271287-fc78-40ff-a74b-7c0a3f86bd6a">

**Comment by prrvchr** [comment]
OAuth2OOo version 1.1.0 is no longer supported, you must migrate to 1.3.4

**Comment by M-Rick** [comment]
Ok so the link on LibreOffice Extensions is outdated.
When you click on 'Download latest', you won't get the very last one, but version 1.1.0.

https://extensions.libreoffice.org/en/extensions/show/oauth2ooo

<img width="1440" alt="Capture d’écran 2024-05-02 à 19 45 24" src="https://github.com/prrvchr/OAuth2OOo/assets/2520948/4ceafeb3-4e6f-45bf-8c67-9cb7a09d25a9">

**Comment by M-Rick** [comment]
This time, I downloaded the right one, but still unworking.

`(com.sun.star.uno.RuntimeException) { { Message = "<class 'ImportError'>: cannot import name 'RequestParameter' from 'oauth2' (/Applications/LibreOffice.app/Contents/Frameworks/LibreOfficePython.framework/Versions/Current/lib/python3.8/site-packages/oauth2/__init__.py), traceback follows\X000a  File \"/Applications/LibreOffice.app/Contents/Resources/pythonloader.py\", line 147, in writeRegistryInfo\X000a    mod = self.getModuleFromUrl( locationUrl )\X000a  File \"/Applications/LibreOffice.app/Contents/Resources/pythonloader.py\", line 102, in getModuleFromUrl\X000a    exec(codeobject, mod.__dict__)\X000a  File \"/Users/aymeric/Library/Application Support/LibreOffice/4/user/uno_packages/cache/uno_packages/lu23015obpk6c.tmp_/OAuth2OOo.oxt/service/OAuth2Service.py\", line 45, in <module>\X000a    from oauth2 import RequestParameter\X000a\X000a at /Users/cloph/build/workdir/UnoApiHeadersTarget/udkapi/normal/com/sun/star/uno/RuntimeException.hdl:24", Context = (com.sun.star.uno.XInterface) @0 } }`

<img width="790" alt="Capture d’écran 2024-05-02 à 19 56 57" src="https://github.com/prrvchr/OAuth2OOo/assets/2520948/29a02e64-1c45-4c50-8a2b-a7f927452648">

**Comment by prrvchr** [comment]
> Ok so the link on LibreOffice Extensions is outdated.

That's right, I don't bother updating it anymore. I'll try to see if I can put a link to github rather than having to update with each new release

It seems that a python package named `oauth2` conflicts with the OAuth2OOo extension.

What's in the directory: `/Applications/LibreOffice.app/Contents/Frameworks/LibreOfficePython.framework/Versions/Current/lib/python3.8/site-packages/oauth2`

Can you try removing this folder `oauth2` (by making a copy) I think it should work...

**Comment by prrvchr** [comment]
On the other hand, it seems that the Python `oauth2` package was installed by LibreOffice (given the path in which it is located: `/Applications/LibreOffice.app/Contents/Frameworks/LibreOfficePython.framework/Versions/Current/lib/python3.8/site-packages`).
Would it be possible for you to confirm this?

**Comment by prrvchr** [comment]
Ok I have just published **OAuth2OOo version 1.3.5** which is installed in the `oauth20` folder instead of `oauth2`.

This prevents any conflict with the python [oauth2](https://pypi.org/project/oauth2/) module, as it seems was happening with your configuration.

I close but you are free to reopen if the problem persists.
